app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -31,101 +31,160 @@ import androidx.compose.material3.ExperimentalMaterial3Api
-                        )
-                    }
+    background {
+
+        val snackbarHostState = remember { SnackbarHostState() }
+
+        Scaffold(
+            modifier = Modifier.semantics {
+                testTagsAsResourceId = true
+            },
+            containerColor = Color.Transparent,
+            contentColor = MaterialTheme.colorScheme.onBackground,
+            contentWindowInsets = WindowInsets(0, 0, 0, 0),
+            snackbarHost = { SnackbarHost(snackbarHostState) },
+            topBar = {
+                // Show the top app bar on top level destinations.
+                val topLevelDestination =
+                    appState.topLevelDestinations[appState.currentDestination?.route]
+                if (topLevelDestination != null) {
+                    NiaTopAppBar(
+                        titleRes = topLevelDestination.titleTextId,
+                        actionIcon = NiaIcons.Settings,
+                        actionIconContentDescription = stringResource(
+                            id = settingsR.string.top_app_bar_action_icon_description
+                        ),
+                        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+                            containerColor = Color.Transparent
+                        ),
+                        onActionClick = { appState.toggleSettingsDialog(true) }
+                    )
                 }
-            ) { padding ->
-                Row(
-                    Modifier
-                        .fillMaxSize()
-                        .windowInsetsPadding(
-                            WindowInsets.safeDrawing.only(
-                                WindowInsetsSides.Horizontal
-                            )
-                        )
-                ) {
-                    if (appState.shouldShowNavRail) {
-                        NiaNavRail(
-                            destinations = appState.topLevelDestinations,
-                            onNavigateToDestination = appState::navigate,
-                            currentDestination = appState.currentDestination,
-                            modifier = Modifier.safeDrawingPadding()
-                        )
-                    }
+            },
+            bottomBar = {
+                if (appState.shouldShowBottomBar) {
+                    NiaBottomBar(
+                        destinations = appState.topLevelDestinations.values,
+                        onNavigateToDestination = appState::navigate,
+                        currentDestination = appState.currentDestination
+                    )
+                }
+            }
+        ) { padding ->
+
+            val isOffline by appState.isOffline.collectAsStateWithLifecycle()
+
+            // If user is not connected to the internet show a snack bar to inform them.
+            val notConnected = stringResource(R.string.for_you_not_connected)
+            LaunchedEffect(isOffline) {
```

Comment:
This is going to be executed every time that `isOffline` changes or this composable is added to the Composition. Do you want to do that or just show it once?

Reply:
This is the same behaviour as we had previously on the "ForYou" screen. 

"or this composable is added to the composition" <- this will only happen when `NiaApp` is recomposed, which iiuc is only during a configuration change, which is what we want, to re-show the snackbar. 

My overall opinion here is that the warning is too aggressive and should probably only be shown once, for a few seconds, but that's a job for another PR: https://github.com/android/nowinandroid/issues/350

Reply:
Please file an issue if you think this should be changed.